### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by nikhil-024**
# Odin version(x.y.z)

### CODE REVIEW

File: proto/gen/go/dream11/od/auth/v1/auth.pb.go
Comment (dhavalmehta07) [inline proto/gen/go/dream11/od/auth/v1/auth.pb.go:1]: Should these be deleted?

File: cmd/describe/environment.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+	},
+	Long: `Describe  environment details`,
+	Run: func(cmd *cobra.Command, args []string) {
+		execute(cmd)
+	},
+}
+
+func init() {
+	environmentCmd.Flags().StringVar(&name, "name", "", "name of the env")
+	environmentCmd.Flags().StringVar(&serviceName, "service", "", "provisioning type of the environment")
```
Comment (dhavalmehta07) [inline cmd/describe/environment.go:34]: description needs to be updated

File: cmd/describe/environment.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+	environmentCmd.Flags().StringVar(&component, "component", "", "cloud provider account name")
```
Comment (dhavalmehta07) [inline cmd/describe/environment.go:35]: same here

File: cmd/describe/environment.go [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+		servicesSummary = append(servicesSummary, serviceMap)
+	}
+	servicesSummaryJSON, err := json.Marshal(servicesSummary)
+	if err != nil {
+		log.Fatal("Failed to marshal services summary: ", err)
+	}
+
+	tableData = append(tableData, []interface{}{
+		*env.Name,
+		"TODO",
```
Comment (dhavalmehta07) [inline cmd/describe/environment.go:118]: ToDo?

File: cmd/describe/environment.go [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+		}
+		if len(svc.Components) > 0 {
+			serviceMap["components"] = svc.Components
+		}
+		servicesSummary = append(servicesSummary, serviceMap)
+	}
+
+	environments = append(environments, map[string]interface{}{
+		"name":                  env.Name,
+		"team":                  "TODO",
```
Comment (dhavalmehta07) [inline cmd/describe/environment.go:154]: todo

File: internal/service/environment.go [hunk 5]
```diff
@@ -108,6 +108,7 @@ func (e *Environment) DeleteEnvironment(ctx *context.Context, request *environme
 	return err
 }
 
+// UpdateEnvironment updates environment
```
Comment (dhavalmehta07) [inline internal/service/environment.go:111]: do we need this comment. Seems obvious :D

Comment (nikhil-024) [reply internal/service/environment.go:111]: github action complains otherwise 🗡️

File: cmd/describe/environment.go (hunk 4 above)
Comment (nikhil-024) [reply cmd/describe/environment.go:154]: this was for team. Will remove it now

File: cmd/describe/environment.go (hunk 3 above)
Comment (nikhil-024) [reply cmd/describe/environment.go:118]: this was for team. Will remove it now

File: cmd/describe/environment.go [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+	Long: `Describe  environment details`,
+	Run: func(cmd *cobra.Command, args []string) {
+		executeEnv(cmd)
+	},
+}
+
+func init() {
+	environmentCmd.Flags().StringVar(&name, "name", "", "name of the environment")
+	environmentCmd.Flags().StringVar(&serviceName, "service", "", "service deployed in this environment")
+	environmentCmd.Flags().StringVar(&component, "component", "", "component of the service")
```
Comment (dhavalmehta07) [inline cmd/describe/environment.go:34]: should we add `deployed in this environment` at the end to be consistent with the service description?

File: cmd/root.go [hunk 7]
```diff
@@ -17,7 +17,7 @@ var RootCmd = &cobra.Command{
 
 func init() {
 	RootCmd.PersistentFlags().StringP("profile", "p", "default", "odin profile")
-	RootCmd.PersistentFlags().StringP("output", "o", "text", "odin profile")
+	RootCmd.PersistentFlags().StringP("output", "o", "json", "odin profile")
```
Comment (dhavalmehta07) [inline cmd/root.go:20]: why do we need this change?

Comment (nikhil-024) [reply cmd/root.go:20]: right, its not needed

**Review by dhavalmehta07 — APPROVED** [review]

**Pull request merged**